src/main.rs

```diff
@@ -45,80 +44,98 @@ struct ValidationError {
  */
 type NamedSchemas = HashMap<String, serde_json::Value>;
 
-/**
- * TopicContext will carry the necessary context into a topic consumer to enable
- * it to properly parse and route messages
- */
-#[derive(Clone, Debug)]
-struct TopicContext {
-    topic: settings::Topic,
-    settings: Arc<settings::Settings>,
-    schemas: Arc<HashMap<String, serde_json::Value>>,
+/// Stores topic metadata including schemas and routing info. Provides helper functions for loading
+/// the desired schema and routing info from a topic name.
+pub struct TopicMap {
+    map: HashMap<String, settings::Topic>,
+    schemas: HashMap<String, serde_json::Value>,
 }
 
-impl TopicContext {
-    /**
-     * Return the schema to use for validating this topic
-     */
-    pub fn schema_to_use(&self, message: &serde_json::Value) -> Option<&serde_json::Value> {
-        match &self.topic.schema {
-            settings::Schema::KeyType { key } => {
-                // Fish out the right sub-value for the key
-                if let Some(schema) = message.get(key) {
-                    // Use the string value assuming we can get it
-                    if let Some(schema) = schema.as_str() {
-                        return self.schemas.get(schema);
-                    }
-                }
-            }
+impl TopicMap {
+    /// Creates a new `TopicMap`.
+    pub fn new(
+        topics: &[settings::Topic],
+        schemas: HashMap<String, serde_json::Value>,
+    ) -> TopicMap {
+        let topics = topics
+            .iter()
+            .map(|t| (t.name.clone(), t.clone()))
+            .into_iter()
```

Comment:
I don't believe an `into_iter()` is necessary on the other side of the map. I had to double-check the documentation, but I believe this is because `Map` implements the [IntoIter trait](https://doc.rust-lang.org/std/iter/struct.Map.html#impl-IntoIterator)